client/tos/snackbar/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Snackbar } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { enableGatewayAfterTosDecline } from '../request.js';
+import './style.scss';
+
+const TosSnackbar = ( { settingsUrl } ) => {
+	const [ visible, setVisible ] = useState( true );
+	const [ busy, setBusy ] = useState( false );
+
+	const enableGateway = async () => {
+		setBusy( true );
+		await enableGatewayAfterTosDecline();
+		window.location = settingsUrl;
```

Comment:
`enableGatewayAfterTosDecline` might throw without any notice. I'd redirect on settings page anyway in this case by setting `window.location` in `finally` block.